torch_xla/amp/autocast_mode.py

```diff
@@ -71,15 +74,23 @@ def __enter__(self):
       self.prev = torch.is_autocast_xla_enabled()  # type: ignore[attr-defined]
       self.prev_dtype = torch.get_autocast_xla_dtype(
       )  # type: ignore[attr-defined]
-      torch.set_autocast_xla_enabled(self._enabled)
-      torch.set_autocast_xla_dtype(self._dtype)
+      if self._xla_bfloat16:
+        torch.set_autocast_enabled(self._enabled)
```

Comment:
if `_xla_bfloat16 `, shouldn't we use `set_autocast_xla_enabled` instead?

Reply:
it will be set by `torch.autocast`, we need to set both.

Reply:
Can we leave a comment here, it is really confusing to see that `if we are using xla_bf16, we need setup upstream autocast to enabled`.